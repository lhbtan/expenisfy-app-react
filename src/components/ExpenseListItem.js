import React from 'react';
import {BrowserRouter, Route, Switch, NavLink, Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ( { id, description, amount, createdAt }) => (
    <div>
    <Link to={`/edit/${id}`}><h3>{description}</h3></Link>
        <p>
        Amount: {numeral(amount / 100).format('$0,0.00')}, 
        Created At: {moment(createdAt).format('MMMM Do, YYYY')}
        </p>       
    </div>
);

export default ExpenseListItem;

