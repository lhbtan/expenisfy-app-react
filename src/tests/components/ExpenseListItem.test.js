import React from 'react'
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem'
import expenses from '../fixtures/expenses';

test('should render an ExpenseListItem with input', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[1]}/>);
    expect(wrapper).toMatchSnapshot();
});
