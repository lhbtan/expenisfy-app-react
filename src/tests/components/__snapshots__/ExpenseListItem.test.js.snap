// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an ExpenseListItem with input 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/2"
  >
    <h3>
      rent
    </h3>
  </Link>
  <p>
    Amount: 
    $22.23
    , Created At: 
    December 28th, 1969
  </p>
</div>
`;
